lib/grandpa/message_handler.go

```diff
@@ -284,22 +284,21 @@ func (h *MessageHandler) verifyCatchUpResponseCompletability(prevote, precommit
 	return nil
 }
 
-func getEquivocatoryVoters(votes []AuthData) (map[ed25519.PublicKeyBytes]struct{}, error) {
+func getEquivocatoryVoters(votes []AuthData) map[ed25519.PublicKeyBytes]struct{} {
 	eqvVoters := make(map[ed25519.PublicKeyBytes]struct{})
-	voters := make(map[ed25519.PublicKeyBytes]int, len(votes))
+	voters := make(map[ed25519.PublicKeyBytes][64]byte, len(votes))
 
 	for _, v := range votes {
-		voters[v.AuthorityID]++
-		switch voters[v.AuthorityID] {
-		case 1:
-		case 2:
```

Comment:
Does not this already take cared of the case when a vote appears twice in the AuthData list?

Reply:
The audit issue and spec say that duplicate votes are not equivocatory, just when votes don't match they are then equivocatory.